avm/res/azure-stack-hci/cluster/tests/e2e/2nodeswitched.defaults/main.test.bicep

```diff
@@ -0,0 +1,107 @@
+
+@description('Optional. The name of the Azure Stack HCI cluster - this must be a valid Active Directory computer name and will be the name of your cluster in Azure.')
+@maxLength(15)
+@minLength(4)
+param name string = 'hcicluster'
+
+@description('Optional. Location for all resources.')
+param location string = deployment().location
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-azure-stack-hci.cluster-${serviceShort}-rg'
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'ashc2nmin'
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+@description('Optional. The password of the LCM deployment user and local administrator accounts.')
+@secure()
+param localAdminAndDeploymentUserPass string = newGuid()
+
+@description('Required. The app ID of the service principal used for the Azure Stack HCI Resource Bridge deployment. If omitted, the deploying user must have permissions to create service principals and role assignments in Entra ID.')
+@secure()
+#disable-next-line secure-parameter-default
+param arbDeploymentAppId string = ''
+
+@description('Required. The service principal ID of the service principal used for the Azure Stack HCI Resource Bridge deployment. If omitted, the deploying user must have permissions to create service principals and role assignments in Entra ID.')
+@secure()
+#disable-next-line secure-parameter-default
+param arbDeploymentSPObjectId string = ''
+
+@description('Required. The secret of the service principal used for the Azure Stack HCI Resource Bridge deployment. If omitted, the deploying user must have permissions to create service principals and role assignments in Entra ID.')
+@secure()
+#disable-next-line secure-parameter-default
+param arbDeploymentServicePrincipalSecret string = ''
+
+@description('Optional. The service principal ID of the Azure Stack HCI Resource Provider. If this is not provided, the module attemps to determine this value by querying the Microsoft Graph.')
+@secure()
+#disable-next-line secure-parameter-default
+param hciResourceProviderObjectId string = ''
+
+var deploymentPrefix = take(uniqueString(namePrefix, serviceShort), 8)
+var deploymentOperations = ['Validate', 'Deploy']
+
+#disable-next-line no-hardcoded-location // Due to quotas and capacity challenges, this region must be used in the AVM testing subscription
+var enforcedLocation = 'southeastasia'
+
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2024-03-01' = {
+  name: resourceGroupName
+  location: enforcedLocation
+}
+
+module hciDependencies 'dependencies.bicep' = {
```

Comment:
The names of all the top-level services should be defined in the below `params` block, including the `dep-` prefix & use of the `serviceShort` & `namePrefix`. This will make troubleshooting a lot easier as it's immediately evident which resources are deployed with which name.
The same comment applies to the other test cases.

Reply:
@AlexanderSehr - Are you looking for the names of the other dependency resources from `avm\utilities\e2e-template-assets\templates\azure-stack-hci\modules\azureStackHCIHost\hciHostDeployment.bicep`? There are ~20 of them, so I am passing the name prefix instead of naming each one here

Reply:
Hey @mbrat2005, I guess you can do that, and naturally I know where you're coming from. I just hope I'll never have to troubleshoot the deployment and then search for the resource names in linked files of linked files 😄